pkg/cosign/verify.go

```diff
@@ -928,6 +946,43 @@ func VerifyBundle(ctx context.Context, sig oci.Signature, rekorClient *client.Re
 	return true, nil
 }
 
+func VerifyTSABundle(ctx context.Context, sig oci.Signature, tsaClient *tsaclient.TimestampAuthority, tsaCertChainPath string) (bool, error) {
+	bundle, err := sig.TSABundle()
+	if err != nil {
+		return false, err
+	} else if bundle == nil {
+		return false, nil
+	}
+
+	b64Sig, err := sig.Base64Signature()
+	if err != nil {
+		return false, fmt.Errorf("reading base64signature: %w", err)
+	}
+
+	fmt.Println("Verifying TSA Bundle")
+
+	sigBytes, err := base64.StdEncoding.DecodeString(b64Sig)
+	if err != nil {
+		return false, fmt.Errorf("reading DecodeString: %w", err)
+	}
+	pemBytes, err := os.ReadFile(filepath.Clean(tsaCertChainPath))
+	if err != nil {
+		return false, fmt.Errorf("error reading certification chain path file: %w", err)
+	}
+	certPool := x509.NewCertPool()
+	ok := certPool.AppendCertsFromPEM(pemBytes)
+	if !ok {
+		return false, fmt.Errorf("error parsing response into Timestamp while appending certs from PEM")
+	}
+
+	err = tsaverification.VerifyTimestampResponse(bundle.SignedRFC3161Timestamp, bytes.NewReader(sigBytes), certPool)
+	if err != nil {
+		return false, fmt.Errorf("unable verifyTSRWithPEM: %w", err)
+	}
+
+	return true, nil
```

Comment:
We also need to verify the certificate expiration as part of verification, like what we do in the rekor bundle - https://cs.github.com/sigstore/cosign/blob/44a7117e0cee495a163c92e9f347efb2d9b747c7/pkg/cosign/verify.go#L907

Reply:
I added support for checking the certificate expiration although I thought we could also have some helper functions in the timestamp-authority to run these validations (meaning the cert and timestamp.Time). wdyt ?